pkg/controller/bucket/bucket_controller.go

```diff
@@ -99,18 +100,32 @@ func (bl *bucketListener) Add(ctx context.Context, obj *v1alpha1.Bucket) error {
 	}
 
 	req := osspec.ProvisionerCreateBucketRequest{
-		BucketName: obj.Spec.BucketRequest.Name,
+		BucketContext: map[string]string{},
+	}
+	if len(obj.Spec.Parameters["BucketInstanceName"]) > 0 {
+		req.BucketName = obj.Spec.Parameters["BucketInstanceName"]
+	} else if len(obj.Spec.Parameters["BucketPrefix"]) > 0 {
+		req.BucketContext["BucketPrefix"] = obj.Spec.Parameters["BucketPrefix"]
```

Comment:
Controller will make sure that BucketInstanceName and BucketPrefix fields exist in the parameters if the BucketRequest for Greenfield and Brownfield has these fields.

Reply:
@brahmaroutu If the BR specifies a prefix value of "" what happens?